gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -87,8 +87,8 @@ public IcebergDataset(String db, String table, IcebergTable icebergTbl, Properti
     this.icebergTable = icebergTbl;
     this.properties = properties;
     this.sourceFs = sourceFs;
-    this.sourceCatalogMetastoreURI = getAsOptionalURI(this.properties, IcebergDatasetFinder.ICEBERG_HIVE_CATALOG_METASTORE_URI_KEY);
-    this.targetCatalogMetastoreURI = getAsOptionalURI(this.properties, TARGET_METASTORE_URI_KEY);
+    this.sourceCatalogURI = getAsOptionalURI(this.properties, IcebergDatasetFinder.ICEBERG_SRC_CATALOG_URI_KEY);
```

Comment:
rather than searching for info in `properties`, let's have the `IcebergCatalog` tell us its identifying URI.  it should know which props it was configured with, and moreover the specific set may vary from catalog to catalog.  definitely let's encapsulate all of that from the code here.

since we don't yet have a dest-side `IcebergCatalog`, ok to leave that one as-is w/ a TODO just above, explaining the plan

Reply:
I have encapsulated the catalog uri information to be a part of `IcebergCatalog` and added the TODO for target catalog!